### PR TITLE
Run utility analyzers under Scanner context

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Related to #4245

### CODE REVIEW

**Comment by pavel-mikula-sonarsource** [comment]
E2E works, I've manually verified it with local SQ.

File: analyzers/tests/SonarAnalyzer.UnitTest/Common/RuleTest.cs [hunk 1]
```diff
@@ -157,6 +157,24 @@ public void DeprecatedRules_AreNotInSonarWay()
             }
         }
 
+        [TestMethod]
+        public void AllRules_DoNotHaveUtilityTag()
+        {
+            foreach (var diagnostic in new RuleFinder().AllAnalyzerTypes.SelectMany(SupportedDiagnostics))
+            {
+                diagnostic.CustomTags.Contains(DiagnosticDescriptorBuilder.UtilityTag).Should().BeFalse();
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Common/RuleTest.cs:165]:
```suggestion
                diagnostic.CustomTags.Should().NotContain(DiagnosticDescriptorBuilder.UtilityTag);
```

File: analyzers/tests/SonarAnalyzer.UnitTest/Common/RuleTest.cs [hunk 2]
```diff
@@ -157,6 +157,24 @@ public void DeprecatedRules_AreNotInSonarWay()
...
+            }
+        }
+
+        [TestMethod]
+        public void UtilityAnalyzers_HaveUtilityTag()
+        {
+            foreach (var diagnostic in new RuleFinder().UtilityAnalyzerTypes.SelectMany(SupportedDiagnostics))
+            {
+                diagnostic.CustomTags.Contains(DiagnosticDescriptorBuilder.UtilityTag).Should().BeTrue();
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Common/RuleTest.cs:174]:
```suggestion
                diagnostic.CustomTags.Should().Contain(DiagnosticDescriptorBuilder.UtilityTag);
```

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**